librad/src/git/p2p/transport.rs

```diff
@@ -231,12 +232,15 @@ where
     io::Error::new(io::ErrorKind::Other, err)
 }
 
-fn block_on<F>(fut: F) -> io::Result<<F::Output as Try>::Ok>
+fn block_on<F>(fut: F) -> io::Result<<F::Output as Try>::Output>
 where
     F: Future,
-    F::Output: Try<Error = io::Error>,
+    F::Output: Try<Residual = io::Result<Infallible>>,
```

Comment:
If this typechecks, the try-trait-v2 proposal is cleary unsound

Reply:
Oh I get it. Not sure why they prefer to introduce such confusing terminology, where "Monad" would clearly do. I do not agree that this pulls it's weight here, feel free to specialise the type to `F: Future<Output = io::Result<T>>`.

Reply:
Because Monads are scary